x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java

```diff
@@ -113,58 +125,116 @@ protected AcknowledgedResponse newResponse() {
+            waitForExistingTaskToComplete(parentTaskClient, request.getJobId(), existingTaskId, listener);
+            return;
+        }
+
+        auditor.info(request.getJobId(), Messages.getMessage(Messages.JOB_AUDIT_DELETING, taskId));
 
         ActionListener<Boolean> markAsDeletingListener = ActionListener.wrap(
                 response -> {
                     if (request.isForce()) {
-                        forceDeleteJob(request, listener);
+                        forceDeleteJob(parentTaskClient, request, listener);
                     } else {
-                        normalDeleteJob(request, listener);
+                        normalDeleteJob(parentTaskClient, request, listener);
                     }
                 },
                 e -> {
-                    if (e instanceof MlMetadata.JobAlreadyMarkedAsDeletedException) {
-                        // Don't kick off a parallel deletion task, but just wait for
-                        // the in-progress request to finish.  This is much safer in the
-                        // case where the job with the same name might be immediately
-                        // recreated after the delete returns.  However, if a force
-                        // delete times out then eventually kick off a parallel delete
-                        // in case the original completely failed for some reason.
-                        waitForDeletingJob(request.getJobId(), MachineLearningField.STATE_PERSIST_RESTORE_TIMEOUT,
-                                ActionListener.wrap(
-                                listener::onResponse,
-                                e2 -> {
-                                    if (request.isForce() && e2 instanceof TimeoutException) {
-                                        forceDeleteJob(request, listener);
-                                    } else {
-                                        listener.onFailure(e2);
-                                    }
-                                }
-                        ));
-                    } else {
-                        listener.onFailure(e);
-                    }
+                    auditor.error(request.getJobId(), Messages.getMessage(Messages.JOB_AUDIT_DELETING_FAILED, e.getMessage()));
+                    listener.onFailure(e);
                 });
 
         markJobAsDeleting(request.getJobId(), markAsDeletingListener, request.isForce());
     }
 
-    @Override
-    protected void masterOperation(DeleteJobAction.Request request, ClusterState state, ActionListener<AcknowledgedResponse> listener) {
-        throw new UnsupportedOperationException("the Task parameter is required");
+    private Optional<Task> findExistingStartedTask(Task currentTask) {
+        return taskManager.getTasks().values().stream().filter(filteredTask ->
+                currentTask.getDescription().equals(filteredTask.getDescription()) &&
```

Comment:
This comparison seems a bit fragile. If we want to do it with task manager (see my other comment above). I would suggest adding a filter for the action name and then if action matches JobDeletion, we can cast it to JobDeletionTask and get job id from the task instead of description.